ls1intum/Artemis
# [Feature] Embed Images in Markdown Editor

### Checklist

### Motivation and Context
Currently only external images can be loaded in the markdown editor. Artemis user can't upload screenshots or similar to visualize their problems / thoughts.

### Description

We allow users to upload images (up to 10MB) using the markdown editor. The functionality is similar to how it is currently in github. You can upload images by copy pasting, dragging and dropping files or clicking the upload box below the editor.

### Steps for Testing
1. Log in to Artemis
2. Navigate to Any place which uses the markdown editor: eg: create an exercise.
3. In the markdown editor, insert your image 
3. Save
4. Now view the exercise (e.g.) The image should be visible.

### Screenshots



Upload box:
<img width="596" alt="Bildschirmfoto 2020-07-16 um 11 11 54" src="https://user-images.githubusercontent.com/51987021/87653445-ce915500-c755-11ea-9c41-243cfb709c56.png">

Uploaded image as markdown:
<img width="577" alt="Bildschirmfoto 2020-07-16 um 11 15 24" src="https://user-images.githubusercontent.com/51987021/87653426-ca653780-c755-11ea-83b3-ee0b7f82ba00.png">

Image embeded in Markdown:
<img width="577" alt="Bildschirmfoto 2020-07-16 um 11 15 32" src="https://user-images.githubusercontent.com/51987021/87653370-bae5ee80-c755-11ea-8c7f-0d6ca76561b2.png">
